tests/dragonfly/cluster_test.py

```diff
@@ -907,272 +972,196 @@ async def test_random_keys():
+    df_local_factory.start_all(instances)
 
-    c_nodes = [node.client() for node in nodes]
-    c_nodes_admin = [node.admin_client() for node in nodes]
+    nodes = [(await create_node_info(instance)) for instance in instances]
+    nodes[0].slots = [(0, 5259)]
+    nodes[1].slots = [(5260, 16383)]
 
-    node_ids = await asyncio.gather(*(get_node_id(c) for c in c_nodes_admin))
+    await push_config(json.dumps(generate_config(nodes)), [node.admin_client for node in nodes])
 
-    config = f"""
-      [
-        {{
-          "slot_ranges": [ {{ "start": 0, "end": LAST_SLOT_CUTOFF }} ],
-          "master": {{ "id": "{node_ids[0]}", "ip": "localhost", "port": {nodes[0].port} }},
-          "replicas": []
-        }},
-        {{
-          "slot_ranges": [ {{ "start": NEXT_SLOT_CUTOFF, "end": 16383 }} ],
-          "master": {{ "id": "{node_ids[1]}", "ip": "localhost", "port": {nodes[1].port} }},
-          "replicas": []
-        }}
-      ]
-    """
+    await check_for_no_state_status([node.admin_client for node in nodes])
 
-    await push_config(
-        config.replace("LAST_SLOT_CUTOFF", "5259").replace("NEXT_SLOT_CUTOFF", "5260"),
-        c_nodes_admin,
+    nodes[0].migrations.append(
+        MigrationInfo("127.0.0.1", nodes[1].instance.admin_port, [(5200, 5259)], nodes[1].id)
     )
 
-    for node in c_nodes_admin:
-        assert await node.execute_command("DFLYCLUSTER", "SLOT-MIGRATION-STATUS") == "NO_STATE"
+    # Push config to source node. Migration will not start until target node gets the config as well.
+    logging.debug("Push migration config to source node")
+    await push_config(json.dumps(generate_config(nodes)), [nodes[0].admin_client])
 
-    migation_config = f"""
-      [
-        {{
-          "slot_ranges": [ {{ "start": 0, "end": LAST_SLOT_CUTOFF }} ],
-          "master": {{ "id": "{node_ids[0]}", "ip": "localhost", "port": {nodes[0].port} }},
-          "replicas": [],
-          "migrations": [{{ "slot_ranges": [ {{ "start": 5200, "end": 5259 }} ]
-                         , "ip": "127.0.0.1", "port" : {nodes[1].admin_port}, "node_id": "{node_ids[1]}" }}]
-        }},
-        {{
-          "slot_ranges": [ {{ "start": NEXT_SLOT_CUTOFF, "end": 16383 }} ],
-          "master": {{ "id": "{node_ids[1]}", "ip": "localhost", "port": {nodes[1].port} }},
-          "replicas": []
-        }}
-      ]
-    """
+    # some delay to check that migration isn't started until we send config to target node
+    await asyncio.sleep(0.2)
 
-    # Push config to source node. Migration will not start until target node gets the config as well.
-    await push_config(
-        migation_config.replace("LAST_SLOT_CUTOFF", "5259").replace("NEXT_SLOT_CUTOFF", "5260"),
-        [c_nodes_admin[0]],
-    )
-    await wait_for_status(c_nodes_admin[0], node_ids[1], "CONNECTING")
-    await wait_for_status(c_nodes_admin[1], node_ids[0], "NO_STATE")
+    await wait_for_status(nodes[0].admin_client, nodes[1].id, "CONNECTING")
```

Comment:
can you rename this function to wait_for_migration_status and add a default timeout as now it is running in for loop with sleep until reaching the expected status

Reply:
do you mean timeout for the whole operation? I don't see any purpose of such timeout for the test because if test is finished by timeout the test is broken.
If you mean sleep time I'm ok to add it

Reply:
I mean that wait_for_migration_status will do the while loop and will break if condition met or timeout reached. If timeout is reached than you should fail the test.
Same as we have in wait_available_async, check_all_replicas_finished